patchwork-registries/src/main/java/net/minecraftforge/registries/ForgeRegistry.java

```diff
@@ -40,17 +41,68 @@
 import net.minecraft.util.registry.DefaultedRegistry;
 import net.minecraft.util.registry.Registry;
 
-public class ForgeRegistry<V extends IForgeRegistryEntry<V>> implements IForgeRegistry<V> {
+import net.patchworkmc.impl.registries.ExtendedVanillaRegistry;
+
+public class ForgeRegistry<V extends IForgeRegistryEntry<V>> implements IForgeRegistryModifiable<V>, IForgeRegistryInternal<V> {
 	public static Marker REGISTRIES = MarkerManager.getMarker("REGISTRIES");
 	private static Logger LOGGER = LogManager.getLogger();
-	private Identifier name;
-	private Registry<V> vanilla;
-	private Class<V> superType;
+	private final Identifier name;
+	private final boolean isVanilla;
+	private final Registry<V> vanilla;
+	private final Class<V> superType;
+	private final Map<Identifier, ?> slaves = Maps.newHashMap();
+	private final CreateCallback<V> create;
+	private final AddCallback<V> add;
+	private final ClearCallback<V> clear;
+	private final RegistryManager stage;
+	private final boolean isModifiable;
+
+	private boolean isFrozen = false;
+
+	/**
+	 * Called by RegistryBuilder, for moded registries.
+	 * @param stage
+	 * @param name
+	 * @param builder
+	 */
+	public ForgeRegistry(RegistryManager stage, Identifier name, RegistryBuilder<V> builder) {
+		this.stage = stage;
+		this.name = name;
+		this.superType = builder.getType();
+		this.create = builder.getCreate();
+		this.add = builder.getAdd();
+		this.clear = builder.getClear();
+		this.isModifiable = builder.getAllowModifications();
 
+		this.vanilla = new ExtendedVanillaRegistry<>(this);
```

Comment:
Maybe use a defaulted registry here if the builder declares default values / keys?

Reply:
defaulted registry is supported now